### ISSUE TITLE
Page navi doesn't work properly

### ISSUE THREAD

**Issue by MkBeeCtrl** [JeremyEnglert/JointsWP#161]
Hello, I am creating simple site for Foundation helping sick kids from Ukraine as a volunteer. I am using your solution, which saved me a lot of time.

During development process I found an issue on page with multiple posts, next pages were automatically created, but when I try to navigate to the second page I am stuck on first, the permalink changes but I am still on first page. Can you please assist me with finding the issue? 

I didn't change any of your core functions, I was looking for an issue in the joints_page_navi function but everything look as it should. Do you have any ideas where is the source of this problem?

Here's the link to the developer edition, directly to the page with an issue:

http://forgotten.beectrl.com/nasze-dzieci/

Best regards, 

Nikolaus

**Comment by Nikolaus91** [comment]
Even with commented page-navi.php I am not able to display in one page all of the posts, however I can find them using Search function.

With commented function I am only able to view posts that previously were on page number 1

**Comment by JeremyEnglert** [comment]
Are you using a custom post type?

**Comment by Nikolaus91** [comment]
Nope, should I?

Already tried different solution from below, but same behaviour. I guess problem is elsewhere :/

`function posts_per_page() {

	if( is_singular() )
		return;

	global $wp_query;

	/** Stop execution if there's only 1 page */
	if( $wp_query->max_num_pages <= 1 )
		return;

	$paged = get_query_var( 'paged' ) ? absint( get_query_var( 'paged' ) ) : 1;
	$max   = intval( $wp_query->max_num_pages );

	/**	Add current page to the array */
	if ( $paged >= 1 )
		$links[] = $paged;

	/**	Add the pages around the current page to the array */
	if ( $paged >= 3 ) {
		$links[] = $paged - 1;
		$links[] = $paged - 2;
	}

	if ( ( $paged + 2 ) <= $max ) {
		$links[] = $paged + 2;
		$links[] = $paged + 1;
	}

	echo '<div class="navigation"><ul>' . "\n";

	/**	Previous Post Link */
	if ( get_previous_posts_link() )
		printf( '<li>%s</li>' . "\n", get_previous_posts_link() );

	/**	Link to first page, plus ellipses if necessary */
	if ( ! in_array( 1, $links ) ) {
		$class = 1 == $paged ? ' class="active"' : '';

		printf( '<li%s><a href="%s">%s</a></li>' . "\n", $class, esc_url( get_pagenum_link( 1 ) ), '1' );

		if ( ! in_array( 2, $links ) )
			echo '<li>…</li>';
	}

	/**	Link to current page, plus 2 pages in either direction if necessary */
	sort( $links );
	foreach ( (array) $links as $link ) {
		$class = $paged == $link ? ' class="active"' : '';
		printf( '<li%s><a href="%s">%s</a></li>' . "\n", $class, esc_url( get_pagenum_link( $link ) ), $link );
	}

	/**	Link to last page, plus ellipses if necessary */
	if ( ! in_array( $max, $links ) ) {
		if ( ! in_array( $max - 1, $links ) )
			echo '<li>…</li>' . "\n";

		$class = $paged == $max ? ' class="active"' : '';
		printf( '<li%s><a href="%s">%s</a></li>' . "\n", $class, esc_url( get_pagenum_link( $max ) ), $max );
	}

	/**	Next Post Link */
	if ( get_next_posts_link() )
		printf( '<li>%s</li>' . "\n", get_next_posts_link() );

	echo '</ul></div>' . "\n";

}`

**Comment by Nikolaus91** [comment]
tried this one to, same stuff here. So there is something wrong on my side. Any tips, got stuck a little bit:

`<div class="navigation"><p><?php posts_nav_link(); ?></p></div>` <-- doesn't work either

**Comment by JeremyEnglert** [comment]
Do either of these help?

https://github.com/JeremyEnglert/JointsWP/pull/104
https://wordpress.org/support/topic/pagination-not-working-shows-the-same-posts-on-first-page

**Comment by JeremyEnglert** [comment]
Also, check out the comments from DaveG on this post: https://github.com/JeremyEnglert/JointsWP/issues/90

He does a great job of explaining the pagination.

**Comment by Nikolaus91** [comment]
I will give it a shot, will replay tommorow. Thanks a lot for your time

**Comment by JeremyEnglert** [comment]
Were you able to get this worked out?

**Comment by Nikolaus91** [comment]
Unfortunately not

2016-03-02 7:52 GMT+01:00 Jeremy Englert <notifications@github.com>:

> Were you able to get this worked out?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JeremyEnglert/JointsWP/issues/161#issuecomment-191096784>
> .
>



-- 
Z poważaniem,

Mikołaj Koropecki


Design & Software
Development House
----------------------------
mob: +48 500 401 730
e-mail: mk@beectrl.com

**Comment by mrscharf** [comment]
I'm not seeing any pagination on the provided demonstration link.

**Comment by Nikolaus91** [comment]
I will turn it on at friday morning, unfortunately I am not able to do it
till tommorow, I had to turn it off

2016-03-02 19:30 GMT+01:00 mrscharf <notifications@github.com>:

> I'm not seeing any pagination on the provided demonstration link.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JeremyEnglert/JointsWP/issues/161#issuecomment-191361177>
> .
>



-- 
Z poważaniem,

Mikołaj Koropecki


Design & Software
Development House
----------------------------
mob: +48 500 401 730
e-mail: mk@beectrl.com

**Comment by Nikolaus91** [comment]
Pagination is turned on

http://lambrefoundation.org/aktualnosci/

2016-03-02 23:29 GMT+01:00 Mikołaj Koropecki <mk@beectrl.com>:

> I will turn it on at friday morning, unfortunately I am not able to do it
> till tommorow, I had to turn it off
>
> 2016-03-02 19:30 GMT+01:00 mrscharf <notifications@github.com>:
>
>> I'm not seeing any pagination on the provided demonstration link.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/JeremyEnglert/JointsWP/issues/161#issuecomment-191361177>
>> .
>>
>
>
>
> --
> Z poważaniem,
>
> Mikołaj Koropecki
>
>
> Design & Software
> Development House
> ----------------------------
> mob: +48 500 401 730
> e-mail: mk@beectrl.com
>
>
>


-- 
Z poważaniem,

Mikołaj Koropecki


Design & Software
Development House
----------------------------
mob: +48 500 401 730
e-mail: mk@beectrl.com

**Comment by mrscharf** [comment]
Try changing this:
`$paged = get_query_var( 'paged' ) ? absint( get_query_var( 'paged' ) ) : 1;`
to this:
`$paged = get_query_var( 'page' ) ? absint( get_query_var( 'page' ) ) : 1;`

**Comment by Nikolaus91** [comment]
Hello, I did a change in a file page-navi.php with the following line:

$paged = intval(get_query_var('paged'));

to 

$paged = get_query_var( 'page' ) ? absint( get_query_var( 'page' ) ) : 1;

no success

I didn't find a line:

$paged = get_query_var( 'paged' ) ? absint( get_query_var( 'paged' ) ) : 1;
[page-navi.zip](https://github.com/JeremyEnglert/JointsWP/files/165813/page-navi.zip)

**Comment by Nikolaus91** [comment]
I have created a developer edition:

http://funlam.beectrl.com/aktualnosci/page/2/

I you wish to change something in admin, let me know I will give you the credentials.

Best regards,

Nicholas

**Comment by mrscharf** [comment]
In your setup, is this a page with a list of posts on it? Are you using a custom query on the page template for this page?

**Comment by Nikolaus91** [comment]
Nope, the page called "*Aktualności*" is a normal page set as Blog front
page in Wordpress settings.

I will send you all the files I am using there in few hours.

2016-03-10 18:46 GMT+01:00 mrscharf <notifications@github.com>:

> In your setup, is this a page with a list of posts on it? Are you using a
> custom query on the page template for this page?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JeremyEnglert/JointsWP/issues/161#issuecomment-194973399>
> .
>



-- 
Z poważaniem,

Mikołaj Koropecki


Design & Software
Development House
----------------------------
mob: +48 500 401 730
e-mail: mk@beectrl.com

**Comment by mrscharf** [comment]
That might help. It seems like a problem I've had on a few projects whenever I wanted to have a page with posts on it (as opposed to just using the usual archive or taxonomy template). To achieve this I used a custom query and loop to display som eposts on the page template. The issue for me was that the joints page navi function only uses the default wp query (the query that pulls up the page). When the page loaded, pagination saw that query and never recognized that I had my own custom query and posts on that page that needed pagination.

My solution was to make the page navi function accept the my custom query object and default to the standard wp query object. I did have to change the 'paged' variable to 'page' or similar to avoid clashing with the default query.

**Comment by Nikolaus91** [comment]
I am using few cutom queries there, it's my first project using WP, I have made few others during last 2 months and I can see that not every approach in this project was well done.

I have attached the whole theme folder. 
It's too big for github so I put it on my Google Drive
https://drive.google.com/file/d/0B6rtKHZbbzVTb3RiWi1WUnFsTTA/view?usp=sharing

I am using custom queries (filters only) for Nasze Dzieci and Projekty. As I recall all of these blog templates use same loop.
